pandas/io/common.py

```diff
@@ -191,14 +192,41 @@ def get_filepath_or_buffer(
 
         .. versionadded:: 1.2.0
 
+    ..versionchange:: 1.2.0
+
+      Returns `mode`. If a path-like object is converted to a file-like object, the
+      returned mode is binary.
+
     Returns
     -------
-    Tuple[FilePathOrBuffer, str, CompressionOptions, bool]
-        Tuple containing the filepath or buffer, the encoding, the compression
-        and should_close.
+    Tuple[FilePathOrBuffer, str, CompressionOptions, bool, Optional[str]]
```

Comment:
can we make a NamedTuple that holds these things (and change where approproate), this is getting too long